### ISSUE TITLE
kong1.3.0 dbless mode fail start: by using basic-auth plugin to set basicauth_credentials

### ISSUE THREAD

**Issue by crazytravel** [Kong/kong#5005]
### Summary

basic-auth plugin on kong 1.3.0 declarative dbless mode start failed. 
Rollback to kong 1.2.0 it works.

nginx: [error] init_by_lua error: ...are/lua/5.1/kong/db/schema/others/declarative_config.lua:423: attempt to index local 'parent_fk' (a nil value)

```yml
basicauth_credentials:
  - consumer: eq-survey
    username: admin
    password: ***********************
```

### Additional Details & Logs

- Kong version (`1.3.0`)
- Kong debug-level startup logs (
```
nginx: [error] init_by_lua error: ...are/lua/5.1/kong/db/schema/others/declarative_config.lua:423: attempt to index local 'parent_fk' (a nil value)
stack traceback:
        ...are/lua/5.1/kong/db/schema/others/declarative_config.lua:423: in function 'get_key_for_uuid_gen'
        ...are/lua/5.1/kong/db/schema/others/declarative_config.lua:457: in function 'generate_ids'
        ...are/lua/5.1/kong/db/schema/others/declarative_config.lua:480: in function 'flatten'
        /usr/local/share/lua/5.1/kong/db/declarative/init.lua:153: in function 'parse_file'
        /usr/local/share/lua/5.1/kong/init.lua:252: in function 'parse_declarative_config'
        /usr/local/share/lua/5.1/kong/init.lua:430: in function 'init'
        init_by_lua:3: in main chunk
```
)
- Docker: kong:1.3.0-alpine

**Comment by andiexer** [comment]
same problem here. try to use jwt plugin. error raises when adding 'key' field

```yaml
jwt_secrets:
 - consumer: test
   algorithm : HS256
   secret: any-secret
   key: any-key
```

**Comment by tisaksen** [comment]
Same here, upgrade from 1.2.1 declarative. Seems like if I delete jwt_secrets and acls it will load fine

**Comment by EvgenyKarataev** [comment]
@andiexer were you able to find a workaround? I am running into exactly the same issue.

**Comment by andiexer** [comment]
@EvgenyKarataev we switched to an older docker image (1.2.2). with this one everything works fine right now

**Comment by EvgenyKarataev** [comment]
@andiexer, thanks for the reply. FYI, I also just found that if you enable the plugin on the consumer directly, it works with 1.3.0, e.g. this worked for me:
`consumers:
- username: test
  jwt_secrets:
  - key: test
    algorithm: HS256
    secret: test
`

**Comment by kikito** [comment]
Hi @crazytravel, thanks for reporting this. I was able to reproduce your issue in 1.3.0, but it seems to have been fixed in `1.4.0rc1`. I also tried it on the `next` branch.

Could you confirm that the problem is gone for you in 1.4.0rc1?
